### ISSUE TITLE
Feature - Catalogue item for F5 integration

### ISSUE THREAD

**Issue by sshipway** [rancher/rancher#3048]
It would be a great idea to have a catalogue item that performs F5 BigIP integration.
In much the same way that the Cloudflare and Route53 applications catch configuration events and update their DNS services, an F5 Service would catch events and manage Virtual IP addresses, balancer pools, and firewalls on an F5 BigIP device using their API.
Host and Service labels could be used to indicate preferences for the integration.

Use cases:
- Add a new host.  If label 'LB' is present, add this host as a wildcard entry in the load balancer pool.
- Remove a host/Modify a host.  Update the pool appropriately.
- Add a service.  If label 'VIP' is present, create a new VIP and backing pool for all instances.  The value of the label indicates the port on the F5 and the public exposed port to link to.
- Add a service/load balancer.  If label 'FW' is present, use BigIP firewall to restrict to only local traffic.

**Comment by bkuhl** [comment]
There's soon to be some F5 BIG-IP Integration - see https://github.com/rancher/community-catalog/tree/master/templates/f5/0 and https://github.com/rancher/external-lb

**Comment by sshipway** [comment]
@bkuhl -- I've taken a look and tried out the new F5 integration.  It looks good for a start, but there are a few more things it needs to support
- multiple partitions (currently, 'Common' partition only)
- specifying the Pool rather than the Server (currently, you specify a server, and it creates its own pool and manages it.  It would be better if I could specify a preexisting pool to be managed)
- adding labelled Hosts to a wildcard Pool (see above use case)
- Being able to specify LB method and irules etc via labels

**Comment by ingardm** [comment]
Being able to specify a (pre existing) pool is essential. A virtual server can have any number of connected pools through irules etc.

**Comment by deniseschannon** [comment]
This is now available in the Catalog.

**Comment by zbialik** [comment]
@deniseschannon The Catalog item still says that it only supports "unpartitioned" F5 BIG-IP installations. Is this still true? 

I have only tested using the f5-bigip-ctlr that is offered by F5 and following [this install doc.](https://clouddocs.f5.com/containers/v2/kubernetes/kctlr-use-bigip-k8s.html#use-bigip-k8s-flannel)

Part of the doc says to create a "dummy" node to act as our gateway to our BIG-IP device. However, when I create the node, it goes from Status: Unknown to Status: NotReady, and then it is removed. I can't tell if my root issue is Kubernetes or Rancher related.

FYI: I actually got the f5-bigip-ctlr working for k8s NodePort services but need to get it working for Ingress/ClusterIP. I've tried each network provider with no success.
